internal/database/sqlcommon/tokenapproval_sql.go

```diff
@@ -187,6 +187,14 @@ func (s *SQLCommon) GetTokenApprovalByProtocolID(ctx context.Context, connector,
 	})
 }
 
+func (s *SQLCommon) GetTokenApprovalByProtocolIDAndPoolID(ctx context.Context, connector, protocolID string, poolID *fftypes.UUID) (*fftypes.TokenApproval, error) {
```

Comment:
Naming thing, but I might suggest deleting `GetTokenApprovalByProtocolID`, calling this method here just `GetTokenApproval`, and calling the by-id method `GetTokenApprovalByID`. I think that's fairly consistent with naming elsewhere and avoids some really long names...